public/controller/data_functions.php

```diff
@@ -278,8 +278,11 @@ function upload_homework($username, $semester, $course, $assignment_id, $homewor
 
     $settings_file = $user_path."/user_assignment_settings.json";
     if (!file_exists($settings_file)) {
-        $json = array("active_assignment"=>$upload_version);
-        file_put_contents($settings_file, json_encode($json));
+        date_default_timezone_set('America/New_York');
+        $version_change_time = date("Y-m-d H:i:s");
+        $history = array(array("version"=>$upload_version, "time"=>$version_change_time));
+        $json = array("active_assignment"=>$upload_version, "history"=>$history);
```

Comment:
Similar to below, I'd prefer the definition for ```$json``` to be one line as it's not improving readability to spread it across multiple:
```
$json = array("active_assignment" => $upload_version, "history" => array("version" => $upload_version",
                                                                                                                      "time" => date("Y-m-d H:i:s")));
```